### ISSUE TITLE
11-bit CAN frames are not handled properly

### ISSUE THREAD

**Issue by pavel-kirienko** [thiemar/mucan#1]
Setup:
- 8devices USB2CAN sending frames to the bus from the interface `can0`
- muCAN is receiving frames from the same bus using the interface `slcan0`
- The bus does not contain other nodes

Initialize the interfaces, then send this:

```
$ cansend can0 12345678#
$ cansend can0 12345678#12345678
$ cansend can0 12345678#1234567812345678
$ cansend can0 123#
$ cansend can0 123#12
$ cansend can0 123#1234
$ cansend can0 123#123456
$ cansend can0 123#12345678
$ cansend can0 123#1234567812
$ cansend can0 123#123456781234
$ cansend can0 123#12345678123456
$ cansend can0 123#1234567812345678
```

It will be reported by muCAN as follows:

```
 (000.000000)  slcan0  12345678   [0]                            ''
 (005.773017)  slcan0  12345678   [4]  12 34 56 78               '.4Vx'
 (014.432340)  slcan0  12345678   [8]  12 34 56 78 12 34 56 78   '.4Vx.4Vx'
 (034.180267)  slcan0  12331234   [5]  62 34 56 78 12            'b4Vx.'
 (036.899465)  slcan0  12341234   [5]  67 84 56 78 12            'g.Vx.'
 (041.404501)  slcan0  12351234   [5]  67 81 26 78 12            'g.&x.'
 (043.072254)  slcan0  12361234   [5]  67 81 23 48 12            'g.#H.'
 (047.321765)  slcan0  12371234   [5]  67 81 23 45 62            'g.#Eb'
 (049.093721)  slcan0  12381234   [5]  67 81 23 45 67            'g.#Eg'
```

SFF frames with less than 3 bytes of payload are ignored, other frames are mangled. The problem could be caused by incorrect SLCAN encapsulation.

**Comment by pavel-kirienko** [comment]
Also, muCAN converts SFF frames to EFF when transmitting.

Sent:

```
 (142.228850)  slcan0       359   [8]  A5 A5 A5 A5 A5 A5 A5 A5   '........'
```

Received:

```
 (142.230507)  can0  00000359   [8]  A5 A5 A5 A5 A5 A5 A5 A5   '........'
```

(Yes, RTT through SocketCAN --> muCAN --> CAN bus --> 8devices USB2CAN --> SocketCAN on a typical desktop Linux is typically around 1~2 ms, which is not bad)

**Comment by bendyer** [comment]
Thanks Pavel! I think this is one of the issues:
https://github.com/thiemar/mucan/blob/master/app/main.c#L147

Will push a fix shortly.

**Comment by bendyer** [comment]
And https://github.com/thiemar/mucan/blob/master/app/main.c#L208-L211

**Comment by bendyer** [comment]
_Should_ be fixed in 593fd48 but I will leave this open until I've tested standard IDs; there may be other issues.

**Comment by bendyer** [comment]
Thanks, that's been fixed. Update instructions: https://github.com/thiemar/mucan/blob/master/README.md
